### PR TITLE
46676207 - VQ-VAE-2 for OASIS brain dataset

### PR DESCRIPTION
**Pull request by Algernon0401**
Hi Shakes and COMP3710 teaching team.

This pull request implements a VQ-VAE-2 and PixelSNAIL model to generate brain MRIs from the OASIS dataset.

The added files are listed below.

- dataset.py: DataLoader of the OASIS dataset
- modules.py: Model of the VQ-VAE-2 and PixelSNAIL
- train.py: Training functions for both VQ-VAE-2 and PixelSNAIL
- predict.py: Functions for image reconstruction and generation
- README.md: Explanation of model architecture, usage, and results

For more information, please read the README.md.

Feel free to let me know if you have any feedback.

Cheers,
Yiqing (Samson) Zhang
46676207

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme Overall:  Good
- Project Overview: Good
- Model: Good
- Data + Preprocessing: Good
- Usage: Good
- Training / Loss Curve(s): no graph(s)
- Result Demonstration: Good
- References: Good

Functionality:
- Reconstruction: Good
- Generation: Good

Code:
- Consistent with Results: Yes
- File Structure: Good
- Commenting: Good
- Commit frequency: Good
- Commit messages: Good

Other Comments:
N/A

**Comment by shakes76** [comment]
# Marking
Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no plots of training etc. -2
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
